src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -53,33 +56,41 @@ ImmutabilityContext context
 			}
 
 			foreach( var member in members ) {
-				// TODO: skip if it has Mutablity.Audited/Mutability.Unaudited.
-
-				// TODO: if it does have one of the above but doesn't need it,
-				//       error out (so we are forced to remove the annotation).
+				if( IsAudited( member ) ) {
+					if( CheckMember( diagnosticSink: _ => { }, member ) ) {
+						m_diagnosticSink(
+							Diagnostic.Create(
+								Diagnostics.UnnecessartyMutabilityAudited,
+								member.GetDeclarationSyntax<MemberDeclarationSyntax>().GetLocation()
+							)
+						);
+					}
+
+					continue;
+				}
 
-				if ( !CheckMember( member ) ) {
+				if ( !CheckMember( m_diagnosticSink, member ) ) {
 					result = false;
 				}
 			}
 
 			return result;
 		}
 
-		private bool CheckMember( ISymbol member ) {
+		private bool CheckMember( DiagnosticSink diagnosticSink, ISymbol member ) {
 			switch( member.Kind ) {
 				case SymbolKind.Field:
-					return CheckField( member as IFieldSymbol );
+					return CheckField( diagnosticSink, member as IFieldSymbol );
 				case SymbolKind.Property:
-					return CheckProperty( member as IPropertySymbol );
+					return CheckProperty( diagnosticSink, member as IPropertySymbol );
 
 				// These member types never contribute to mutability:
 				case SymbolKind.Method:
 				case SymbolKind.NamedType:
 					return true;
 
 				case SymbolKind.Event:
-					m_diagnosticSink(
+					diagnosticSink(

```

Comment:
```csharp
// By default raise an alarm (in case we missed something, or
				// if there are new unsupported language features.)
				default:
					m_diagnosticSink(
						Diagnostic.Create(
							Diagnostics.UnexpectedMemberKind,
							member.GetDeclarationSyntax<MemberDeclarationSyntax>().GetLocation(),
							member.Name,
							member.Kind
						)
					);
```

I left this one as `m_diagnosticSink`... but maybe it makes sense to switch too in order to allow someone to use a thing and mark it with `NeedToMakeTheAnalyzerCosiderThisSafe`?